src/main/java/com/example/librarymanagementsystem/model/Reservation.java

```diff
@@ -0,0 +1,97 @@
+package com.example.librarymanagementsystem.model;
+
+import java.time.LocalDate;
+
+public class Reservation {
+
+    private int reservationId;
+    private int resourceId;
+    private int userId;
+    private Status reservationStatus;
+    private LocalDate reservationDate;
+    private int queuePosition;
+
+    /**
+     * Instantiates a new Reservation.
+     *
+     * @param reservationId     the reservation id
+     * @param resourceId            the resource id
+     * @param userId            the user id
+     * @param reservationStatus the reservation status
+     * @param reservationDate   the reservation date
+     * @param queuePosition     the queue position
+     */
+    public Reservation(int reservationId, int resourceId, int userId, Status reservationStatus, LocalDate reservationDate, int queuePosition) {
+        this.reservationId = reservationId;
+        this.resourceId = resourceId;
+        this.userId = userId;
+        this.reservationStatus = reservationStatus;
+        this.reservationDate = reservationDate;
+        this.queuePosition = queuePosition;
+    }
+
+    /**
+     * Instantiates a new Reservation.
+     *
+     * @param resourceId            the resource id
+     * @param userId            the user id
+     * @param reservationStatus the reservation status
+     * @param reservationDate   the reservation date
+     * @param queuePosition     the queue position
+     */
+    public Reservation(int resourceId, int userId, Status reservationStatus, LocalDate reservationDate, int queuePosition) {
+        this.resourceId = resourceId;
+        this.userId = userId;
+        this.reservationStatus = reservationStatus;
+        this.reservationDate = reservationDate;
+        this.queuePosition = queuePosition;
+    }
+
```

Comment:
Instead of defining strict constructors for other parameters, use builders (Builder Design Pattern) to make it more flexible, readable and reducing error encounters. Look into that and see how it works.